File: Add Code Here/JAVA/binary_search.java
public class binary_search {
    public static void main(String[]args)
    {
        int [] arr = {-12,34,21,45,56,76,22};
        int target = 34;
        int ans = Binarysearch(arr,target);
        System.out.println(ans);

    }
    static int Binarysearch (int [] arr, int target)
    {
        int start = 0;
        int end = arr.length-1;

        while(start <= end)
        {
            int mid = start + (end-start) / 2;
            if(target <arr[mid])
            {
                end = mid -1;
            }
            else if(target > arr[mid])
            {
                start = mid + 1;
            }
            else {
                return mid;
            }
        }
        return -1;
    }
    }

File: Add Code Here/TypeScript/Binary Search.ts
function binary_search(x: number[], f: number): void {
  let l: number = 0;
  let h: number = x.length - 1;
  let flag: boolean = false;

  while (l < h) {
    let mid: number = Math.floor((l + h) / 2);
    if (x[mid] === f) {
      flag = true;
      break;
    } else {
      if (x[mid] < f) {
        l = mid + 1;
      } else if (x[mid] > f) {
        h = mid - 1;
      }
    }
  }

  if (flag) {
    console.log(f + ' has been found!');
  } else {
    console.log(f + ' is not found!');
  }
}

const arr: number[] = [1, 3, 5, 6, 7, 77, 88, 99];
const find: number = 1;
binary_search(arr, find);
